grpc/diatheke.proto

```diff
@@ -281,10 +303,40 @@ message PushTextRequest {
   string text = 2;
 }
 
+// Request to change the current story of a session.
+message StoryRequest {
+  // ID of the session that will have its story changed.
+  string session_id = 1;
+
+  // ID of the story to switch to. This ID is defined by the model
+  // used to create the session.
+  string story_id = 2;
+
+  // Parameters to set in session memory before executing the specified
+  // story. Some stories in the model may make assumptions about which
+  // parameters have already been defined, so it is important to be
+  // familiar with the model requirements for any given story.
+  map<string, string> parameters = 3;
+
+  // If true, the given story will not be executed until the session
+  // completes the current stories and returns back to the main story.
+  // If false, the current story in the session will be immediately
+  // interrupted to execute the specified story.
+  bool wait_for_start = 4;
+
+  // If true, once the given story has finished, Diatheke will return
+  // the session to the place in the model where it was when this request
+  // was received. This is useful when the change in story represents a
+  // temporary interruption. If false, Diatheke will simply continue
+  // from the given story without trying to go back to its prior state,
+  // which is useful to make a permanent state change.
+  bool temporary = 5;
```

Comment:
If the session returns to the original place it the temporary story was initiated from, will it retain values of any parameters that it had before jumping into the temporary story?

Reply:
That is a great question that I haven't fully decided yet. I lean towards restoring the session memory to its prior state if it is temporary (I don't think that would be too difficult), but I can imagine scenarios where maybe I don't want to. What do you think would be best here?

Reply:
I am not sure, if it's used for things like alerts (an alert in the middle of filling up some form for example), then I would say we should retain the parameter values once we return. I am not sure what the best thing to do here would be; I am wondering whether we can expand `wait_for_start` to control this behavior as well. (waiting for start implies we don't have any param values to restore, otherwise we always store values ?)

Reply:
I don't think `wait_for_start` is the right way to control this behavior. That would prevent story changes that need to happen immediately (like an alert/alarm) from being able to change or restore memory. However, now that I think about it more, we should probably define the session "state" as the current location in the model AND the current parameters in memory. In which case, it would make sense for parameters to be restored based on the `temporary` flag. Let's do that - if `temporary` is true, we will restore session memory to its prior state, but if it is false we don't worry about it. I'll add some comments here to make that explicit.